### PR TITLE
Add Guava PreCondition validations to setters

### PR DESCRIPTION
**Pull request by martinremis**
## Summary
* Add Guava PreCondition validations to setters
## Trello
https://trello.com/c/lW0doehf/29-agregar-validaci%C3%B3n-de-precondiciones-a-los-setters

### CODE REVIEW

File: src/main/java/wolox/training/models/Book.java [hunk 1]
```diff
@@ -94,63 +95,63 @@ public String getAuthor() {
   }
 
   public void setAuthor(String author) {
-    this.author = author;
+    this.author = Preconditions.checkNotNull(author, ErrorMessages.CANNOT_BE_NULL, "author");
```
Comment (rabinovichm) [inline src/main/java/wolox/training/models/Book.java]: The precondition should be checked before assigning the value to the object :yum:

File: src/main/java/wolox/training/models/Book.java [hunk 2]
```diff
@@ -94,63 +95,63 @@ public String getAuthor() {
...
+    this.year = Preconditions.checkNotNull(year, ErrorMessages.CANNOT_BE_NULL, "year");
   }
 
   public String getPages() {
     return pages;
   }
 
   public void setPages(String pages) {
-    this.pages = pages;
+    this.pages = Preconditions.checkNotNull(pages, ErrorMessages.CANNOT_BE_NULL, "pages");
```
Comment (rabinovichm) [inline src/main/java/wolox/training/models/Book.java]: The number of pages should be greater than cero

File: src/main/java/wolox/training/models/User.java [hunk 3]
```diff
@@ -68,32 +70,28 @@ public User(String username, String name, LocalDate birthdate, List<Book> books)
...
+    this.name = Preconditions.checkNotNull(name, ErrorMessages.CANNOT_BE_NULL, "name");
   }
 
   public LocalDate getBirthdate() {
     return birthdate;
   }
 
   public void setBirthdate(LocalDate birthdate) {
-    this.birthdate = birthdate;
+    this.birthdate = Preconditions.checkNotNull(birthdate, ErrorMessages.CANNOT_BE_NULL, "birthdate");
```
Comment (rabinovichm) [inline src/main/java/wolox/training/models/User.java]: The birthdate must be earlier than today

**Review by rabinovichm — CHANGES_REQUESTED** [review]
Now that you have preconditions to set object's properties, the objects' constructors should use the setters instead of assigning the values using _this.property = value_

File: src/main/java/wolox/training/models/Book.java [hunk 4]
```diff
@@ -94,6 +96,7 @@ public String getAuthor() {
   }
 
   public void setAuthor(String author) {
+    Preconditions.checkNotNull(author, StatusMessages.CANNOT_BE_NULL, "author");
```
Comment (rodrigofranc) [inline src/main/java/wolox/training/models/Book.java:99]: Any string should be not empty

File: src/main/java/wolox/training/models/Book.java [hunk 5]
```diff
@@ -110,6 +114,7 @@ public String getTitle() {
   }
 
   public void setTitle(String title) {
+    Preconditions.checkNotNull(title, StatusMessages.CANNOT_BE_NULL, "title");
```
Comment (rodrigofranc) [inline src/main/java/wolox/training/models/Book.java:119]: Any string should be not empty

File: src/main/java/wolox/training/models/User.java [hunk 6]
```diff
@@ -79,6 +76,7 @@ public String getName() {
   }
 
   public void setName(String name) {
+    Preconditions.checkNotNull(name, StatusMessages.CANNOT_BE_NULL, "name");
```
Comment (rodrigofranc) [inline src/main/java/wolox/training/models/User.java:80]: Any string should be not empty

File: src/main/java/wolox/training/models/Book.java [hunk 7]
```diff
@@ -150,6 +161,7 @@ public String getIsbn() {
   }
 
   public void setIsbn(String isbn) {
+    Preconditions.checkNotNull(isbn, StatusMessages.CANNOT_BE_NULL, "isbn");
```
Comment (rodrigofranc) [inline src/main/java/wolox/training/models/Book.java:169]: Any string should be not empty

**Review by rodrigofranc — CHANGES_REQUESTED** [review]

**Review by rabinovichm — APPROVED** [review]

**Review by rodrigofranc — APPROVED** [review]

**Pull request merged**